src/AzureAuth/ExceptionListToStringConverter.cs

```diff
@@ -51,39 +52,58 @@ public static List<string> ExtractCorrelationIDsFromException(IList<Exception> e
         }
 
         /// <summary>
-        /// Executes the convertion of exceptions to a string.
+        /// Executes the convertion of exceptions to a JSON string.
         /// </summary>
-        /// <param name="exceptions">The exceptions.</param>
-        /// <returns>The <see cref="string"/>.</returns>
+        /// <param name="exceptions">List of exceptions.</param>
+        /// <returns>JSON format of exception list.</returns>
         public static string Execute(IEnumerable<Exception> exceptions)
         {
             if (exceptions == null || exceptions.Count() == 0)
             {
                 return null;
             }
 
-            return string.Join("\n", exceptions.Select(SingleLineException));
+            List<CustomException> customExceptions = new List<CustomException>();
+            foreach (Exception exception in exceptions)
+            {
+                var customException = new CustomException(exception);
+                customExceptions.Add(customException);
+            }
+
+            return JsonSerializer.Serialize(customExceptions);
         }
+    }
 
+    /// <summary>
+    /// CustomException class with only select properties of Exception class.
+    /// </summary>
+    internal class CustomException
+    {
         /// <summary>
-        /// Converts exceptions to a single string.
+        /// Initializes a new instance of the <see cref="CustomException"/> class.
         /// </summary>
-        /// <param name="ex">The exceptions.</param>
-        /// <returns>The <see cref="string"/>.</returns>
-        private static string SingleLineException(Exception ex)
+        /// <param name="exception">Exception.</param>
+        public CustomException(Exception exception)
         {
-            if (ex != null)
+            if (exception?.Message != null)
             {
-                var message = ex.Message
-                                .Replace("\n", string.Empty)
-                                .Replace("\r", string.Empty)
-                                .Replace("\t", string.Empty);
-                return $"{ex.GetType()}: {message}";
+                var singleLineMessage = exception.Message
+                                    .Replace("\n", string.Empty)

```

Comment:
Now that we are formatting the whole list as a JSON array, we don't need to replace newlines, but it will still make querying easier if we replace `\r` with nothing. This effectively says, translate `\r\n` -> `\n` so we preserve the newline, but in a way that will make querying or splitting on them easy if needed. We can leave `\t` in the message.

Reply:
@kyle-rader I was just trying to understand the difference between `\r` and `\n`. Looks like `\r` is a new line character on mac, `\n` is a new line character on linux and `\r\n` is on windows. So should I just leave all of them as is along with `\t`?

Reply:
`\r` is the "Carriage return" (symbolic of a typewriter moving back to the start o the line or returning the carriage back to the starting position). In Computers this is like back to start of the current line, opposed to a new line or "line feed" - also from type writers literally feeding the paper up to make a new line.

Today, `\r` is only a valid line ending on it's own on obscure systems.  Mac and Linux both use `\n` while Windows still uses `\r\n`. (Also seen as `LF` pronounced L-F and `CRLF`, pronounced "curl-if").

Reply:
Thanks, @kyle-rader!